src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java

```diff
@@ -67,6 +75,7 @@ public String getCompanyNumber() {
 
     public void setCompanyNumber(String companyNumber) {
         this.companyNumber = companyNumber;
+        this.setLinks();
```

Comment:
Doing this just doesn't look right. I'll have a look at code to understand the issue and see if I can see an alternative solution.

Reply:
Ideally we'd be able to add a little private method to the objection mapper "generateDTOLinks"
Then call objectionResponseDTO.setLinks(generateDTOLinks(ObjectionEntity); as part of the mapping.

Bu that's all abstracted away into mapstruct, and not sure if there's a way we could do that